fleet_adapter_mir/fleet_adapter_mir/fleet_adapter_mir.py

```diff
@@ -55,229 +68,138 @@ def sanitise_dict(dictionary, inplace=False, recursive=False):
-                node=handle_data['node'],
-                rmf_graph=handle_data['graph'],
-                robot_traits=handle_data['robot_traits'],
-                robot_state_update_frequency=rmf_config.get('robot_state_update_frequency', 1),
-                dry_run=dry_run
-        )
-        robot.mir_api =  MirAPI(prefix,headers)
-        robot.transforms = handle_data['transforms']
-        robot.rmf_map_name = rmf_config['start']['map_name']
-
-        if not dry_run:
-        #    with MiRRetryContext(robot):
-            _mir_status = robot.mir_api.status_get()
-            robot.mir_name = _mir_status["robot_name"]
-
-            robot.load_mir_missions()
-            robot.load_mir_positions()
-
-            # Check that the MiR fleet has defined the RMF move mission,
-            # that this adapter will use repeatedly with varying parameters.
-            rmf_move_mission = mir_config['rmf_move_mission']
-            assert rmf_move_mission in robot.mir_missions, \
-                (f'RMF move mission [{rmf_move_mission}] not yet defined as a mission in MiR fleet')
-            robot.mir_rmf_move_mission = rmf_move_mission
-
-            if 'dock_and_charge_mission' in mir_config:
-                dock_and_charge_mission = mir_config['dock_and_charge_mission']
-                assert dock_and_charge_mission in robot.mir_missions, \
-                (f'Dock and charge mission [{dock_and_charge_mission}] not yet defined as a mission in MiR fleet')
-                robot.mir_dock_and_charge_mission = dock_and_charge_mission
 
-        else:
-            robot.mir_name = "DUMMY_ROBOT_FOR_DRY_RUN"
+def create_fleet(
+        fleet_config, config_yaml, cmd_node, rmf_missions) -> FleetAdapterMiR:
+    """Create RMF Adapter and fleet handle"""
+    for level, coords in \
+            config_yaml['conversions']['reference_coordinates'].items():
+        tf = compute_transforms(level, coords, cmd_node)
+        fleet_config.add_robot_coordinates_transformation(level, tf)
 
-        robots[robot.name] = robot
+    # RMF_CORE Fleet Adapter: Manages delivery or loop requests
+    adapter = rmf_adapter.Adapter.make(
+        config_yaml['node_names']['rmf_fleet_adapter'])
+    assert adapter, ("Adapter could not be init! "
+                     "Ensure a ROS2 scheduler node is running")
 
-        # OBTAIN PLAN STARTS ==================================================
-        start_config = rmf_config['start']
+    cmd_node.declare_parameter('server_uri', '')
+    server_uri = cmd_node.get_parameter(
+        'server_uri'
+    ).get_parameter_value().string_value
+    if server_uri == '':
+        server_uri = None
+
+    fleet_config.server_uri = server_uri
+    fleet_handle = adapter.add_easy_fleet(fleet_config)
+
+    conversions = config_yaml['conversions']
+    update_frequency = config_yaml['rmf_fleet']['robot_state_update_frequency']
+    debug = config_yaml['rmf_fleet']['debug']
+    plugin_config = config_yaml.get('plugins')
```

Comment:
I don't see anywhere that we're calling [`add_performable_action`](https://github.com/open-rmf/rmf_ros2/blob/1db8a6559c58994f5024138daa1d04b2bec2812a/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/FleetUpdateHandle.hpp#L192-L205). If I remember correctly the fleet adapter will reject all tasks that include an action which we haven't added to the fleet, so robots will only be able to do patrols if we don't call that function.

I recommend iterating through all the plugins and adding performable actions for each action listed in the `actions:` list of each plugin.

Reply:
I initially intended for users to duplicate their plugin actions in the fleet config to add them in as performable actions, but this is a much better solution that minimizes user diligence. Added in [49724d8](https://github.com/open-rmf/fleet_adapter_mir/pull/36/commits/49724d88bc58dcfa864733867062d30aee7fc8a7)

Reply:
It occurs to me that a user might sometimes want a custom [consideration callback](https://github.com/open-rmf/rmf_ros2/blob/dd91566c7e9483bfa0d053d9ad1846b5fc725ae6/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/FleetUpdateHandle.hpp#L205), in which case they'll need to call `add_performable_action` manually.

We could consider having a config parameter like `auto_add_action: False` to give the user the chance to set this manually while their factory is being built. But this isn't important for this PR, so let's just open an issue ticket for this and leave it for later if anyone wants it.